sklearn/metrics/_scorer.py

```diff
@@ -355,6 +357,27 @@ def set_score_request(self, **kwargs):
             self._metadata_request.score.add_request(param=param, alias=alias)
         return self
 
+    def get_metadata_routing(self):
+        """Get requested data properties.
+
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        Returns
+        -------
+        request : MetadataRequest
+            A :class:`~sklearn.utils.metadata_routing.MetadataRequest` instance.
+        """
+        if hasattr(self, "_metadata_request"):
+            requests = get_routing_for_object(self._metadata_request)
+        else:
+            requests = self._get_default_requests(
+                score_method=self._score_func,
+                ignore_params={"y_true", "y_pred"},
```

Comment:
we explicitly pass the `_score_func` so that the right metadata can be deducted from its signature.

And we need to ignore `y_true` and `y_pred` in that process.